src/inmanta/env.py

```diff
@@ -687,6 +635,19 @@ def install_for_config(
         upgrade_strategy: PipUpgradeStrategy = PipUpgradeStrategy.ONLY_IF_NEEDED,
         paths: List[LocalPackagePath] = [],
     ) -> None:
+        """
+        Perform a pip install in this environment, according to the given config
+
+        :param requirements: which requirements to install
+        :param paths: which paths to install
+        :param config: the pip config to use
+        :param constraint_files: pass along the following constraint files
+        :param upgrade: make pip do an upgrade
+        :param upgrade_strategy: what upgrade strategy to use
+
+        limitation:
+         - When upgrade is false, we check if the requirements are installed, but we don't take the constraints into account.
```

Comment:
I don't think this is entirely accurate / ambiguous. If I understand correctly:
1. It applies only to the pre-install check (which may skip the install), while "don't take the constraints into account" seems to imply we just ignore the constraints when upgrade is false.
2. This is only true for the `ActiveEnv` subclass.

Therefore I think it would be more appropriate to mention that we "don't guarantee" rather than "don't take into account". This would resolve both issues with this statement.

```suggestion
         - When upgrade is false, if requirements are already installed constraints from constraint files may not be verified.
```